internal/cmd/agent/root.go

```diff
@@ -125,6 +124,15 @@ func (a *FleetAgent) Run(cmd *cobra.Command, args []string) error {
 		RenewDeadline: *leaderOpts.RenewDeadline,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(ctx context.Context) {
+				// Handle agent registration.
+				r := &Register{}
+				r.Namespace = a.Namespace
+				err := r.RegisterAgent(ctx)
+				if err != nil {
```

Comment:
nit: this could perhaps be simplified a bit, e.g.:
```suggestion
				r := Register{Namespace: a.Namespace}
				if err := r.RegisterAgent(ctx); err != nil {
```

Reply:
AFAIK, that wouldn't work, since the `Namespace` attribute is part of the `UpstreamOptions` that is embedded into `Register`, which would rather make it look like this:

```diff
-               r := &Register{}
-               r.Namespace = a.Namespace
+               r := &Register{
+                   UpstreamOptions: UpstreamOptions{
+                       Namespace:  a.Namespace,
+                   },
+               }
```

I do not mind making it look like that. Perhaps it's clearer that way.

Reply:
Oh I see. Nevermind then, the current version is more concise.